packages/wallet/test/services/TransactionsTracker.test.ts

```diff
@@ -33,13 +45,45 @@ describe('TransactionsTracker', () => {
         store,
         tipBlockHeight$
       }).transactionsSource$;
-      expect(await firstValueFrom(provider$)).toEqual(queryTransactionsResult);
+      expect(await firstValueFrom(provider$)).toEqual(queryTransactionsResult.pageResults);
       expect(store.setAll).toBeCalledTimes(1);
-      expect(store.setAll).toBeCalledWith(queryTransactionsResult);
+      expect(store.setAll).toBeCalledWith(queryTransactionsResult.pageResults);
+    });
+
+    it('emits entire transactions list resolved by ChainHistoryProvider', async () => {
+      const pageSize = PAGE_SIZE;
+      const totalTxsCount = 30;
```

Comment:
It would be nice if you calculated this one, so it autoupdates if PAGE_SIZE changes:
```diff
- const totalTxsCount = 30;
+ const secondPageSize = 5;
+ const totalTxsCount = pageSize + secondPageSize;

Reply:
Updated in 5ab9d3ab3c3777f603b9e6f8630578b796b8a06d @mirceahasegan